test/integration/uk/gov/hmrc/cdsimportsdds/repositories/DeclarationRepositoryTest.scala

```diff
@@ -62,4 +61,23 @@ class DeclarationRepositoryTest extends WordSpec
     }
   }
 
+  "findByEori" should {
+    val declarations = Seq(
+      anImportsDeclaration.copy(id = "1", eori = "GB1111"),
+      anImportsDeclaration.copy(id = "2", eori = "GB3333"),
+      anImportsDeclaration.copy(id = "3", eori = "GB2222"),
+      anImportsDeclaration.copy(id = "4", eori = "GB3333")
+    )
+
+    "return all declarations for the given EORI" in {
+      repository.collection.insert.many(declarations)
+      repository.findByEori("GB3333").map(_.map(_.id)).futureValue shouldBe Seq("2", "4")
```

Comment:
`.map(_.map(_.id)`? why not assert on the `futureValue` here?

Reply:
thanks, have updated now